contest/remote/vmksft-p.py

```diff
@@ -135,6 +171,22 @@ def _vm_thread(config, results_path, thr_id, in_queue, out_queue):
         out_queue.put({'prog': prog, 'test': test_name, 'file_name': file_name,
                        'result': result, 'time': (t2 - t1).seconds})
 
+        parse_tap = config.getboolean('ksft', 'parse_tap', fallback=False)
+        if parse_tap:
+            parse_nested_tests = config.getboolean('ksft', 'parse_nested_tests',
+                                                   fallback=False)
+            tests = _parse_tap(vm.log_out, parse_nested_tests)
+
+            for r_name, r_result in tests:
+                # skip the main test, already covered above
+                if r_name == f'selftests: {target}: {prog}':
```

Comment:
Sorry, I really had to go last night, and I didn't have the opportunity to write a commit message. I pushed the modification just in case, but it would have been better with a proper commit message :)

The TAP parsing is done after what is already in place, it doesn't replace it. I didn't want to replace it, not to loose the history if the name is different. But also because the code above checks if something "bad" happened. So here, we skip the "main" test, because it is already handled.

Should I do something different?

Reply:
Now that I'm thinking about it: here, we know that there will be only one selftest being launched, so only one selftest result, no? I mean, we will always have a single "main" test, always this at the beginning of the stdout file:

```
TAP version 13
1..1
```

So maybe, we only need to support the parsing of nested tests in the comments, no?